### PR TITLE
Returns processed/indexed events

### PR DESCRIPTION
**Pull request by jochenschneider**
After looking into this OTA-4163 needs some work outside of the scope of OTA-4320, so this is just part of OTA-4320.

Copied over the code and test from API provider. Created a v2 of the `devices/$deviceId/events` endpoint. The api provider code will be removed later once OTA-4320 is done.

Signed-off-by: Jochen Schneider <j.schneider@here.com>

### CODE REVIEW

File: src/main/scala/com/advancedtelematic/ota/deviceregistry/DeviceResource2.scala [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  implicit val apiUpdateStatusCodec = deriveCodec[ApiDeviceEvents]
+}
+
+class DeviceResource2(namespaceExtractor: Directive1[AuthedNamespaceScope], deviceNamespaceAuthorizer: Directive1[DeviceId])
+                     (implicit db: Database, ec: ExecutionContext) {
+
+  val eventJournal = new EventJournal()
+
+  def findUpdateEvents(namespace: Namespace, deviceId: DeviceId, correlationId: Option[CorrelationId]): Future[ApiDeviceEvents] = async {
+    val indexedEvents = await(eventJournal.getIndexedEvents(deviceId, correlationId))
```
[the remark this answers is not in the document]
Comment (jochenschneider) [reply src/main/scala/com/advancedtelematic/ota/deviceregistry/DeviceResource2.scala]: I guess that's a matter of taste. I don't mind the async/await code and am not a big fan of nested `map`s. But to be honest I just copied that code over and left it like it was.

**Comment by houcros** [comment]
> The api provider code will be removed later once OTA-4320 is done.
Do you mean here, the code under the route `api/v2`? Or from the new api-provider service?

I am a bit afraid of what could happen if we realise we need to continue adding routes here for the api-provider. We don't actually version our APIs yet, but we could consider using a special `api/v1provider` or something like that instead of `api/v1`, in case we do want to bump the whole API v2 some sunny day?
But dunno, guess that's a big complicated topic...

File: src/main/scala/com/advancedtelematic/ota/deviceregistry/DeviceResource2.scala [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import scala.concurrent.{ExecutionContext, Future}
+
+object DeviceResource2 {
+  import io.circe.generic.semiauto._
+
+  type ApiUpdateId = CorrelationId
+
+  type ApiDeviceUpdateEventName = IndexedEventType
+
+  case class ApiDeviceEvent(ecuId: Option[EcuIdentifier], updateId: Option[ApiUpdateId], name: ApiDeviceUpdateEventName,
```
[the remark this answers is not in the document]
Comment (jochenschneider) [reply src/main/scala/com/advancedtelematic/ota/deviceregistry/DeviceResource2.scala]: I guess it's obvious enough and more concise this way, but the truth is again that we gave the [API](https://github.com/advancedtelematic/ota-device-registry/blob/7494a0f63384b64c386c2fd74fb1e2f5db74006b/src/main/resources/api-definition.yml#L214) out already and I don't think it's worth changing.

File: src/main/scala/com/advancedtelematic/ota/deviceregistry/DeviceResource2.scala [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                            receivedTime: Instant, deviceTime: Instant)
+
+  case class ApiDeviceEvents(deviceUuid: DeviceId, events: Vector[ApiDeviceEvent])
```
[the remark this answers is not in the document]
Comment (jochenschneider) [reply src/main/scala/com/advancedtelematic/ota/deviceregistry/DeviceResource2.scala]: The API already semi-public by now.

File: src/main/scala/com/advancedtelematic/ota/deviceregistry/DeviceResource2.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (houcros) [reply src/main/scala/com/advancedtelematic/ota/deviceregistry/DeviceResource2.scala]: Ah ok, yes if it's out this is not the place to change the names...

**Comment by jochenschneider** [comment]
I meant I want to remove the endpoints which are currently under device-registry/api-provider/api/v1alpha  and move the ones which are really new under the main device-registry endpoints. The v2/events endpoint I created here is technically a "better" versions of v1/events and I didn't want to call it v1/indexedEvents or something.

**Comment by houcros** [comment]
Ah I see. Yes, in that case I think it makes sense to deprecate v1 events in favour of v2 events.
The only thing is that v1 events returns a `payload.success` flag that v2 doesn't. But I have the suspicion that the UI is not actually using it and we might not need it. If I'm wrong, however, it might be a bit more difficult to get rid of v1 events.

File: src/main/scala/com/advancedtelematic/ota/deviceregistry/DeviceResource2.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (houcros) [reply src/main/scala/com/advancedtelematic/ota/deviceregistry/DeviceResource2.scala]: Guess my question comes too late then 😄

**Comment by jochenschneider** [comment]
Ah ok I see in app
```
ota-plus-web/app/reactapp/src/components/device/InstallationEvents.jsx
61:              preparedEvents.ecuDownloadCompleted.status = el.payload.success
76:              preparedEvents.ecuInstallationCompleted.status = el.payload.success
```

**Pull request merged**